### DISCUSSION TITLE
Model takes too much time to train

### DISCUSSION THREAD

**Discussion by SailSabnis** [mrdbourke/pytorch-deep-learning#1037] [Q&A] [answered]
Hi all,

I wrote the exact same code as Daniel and trying to run it on 2 epochs. But, it takes forever. I am on CPU but still it takes unusually long time. Is this normal?

<img width="609" alt="Screenshot 2024-08-08 at 12 59 13 PM" src="https://github.com/user-attachments/assets/a039b386-f8e9-4af7-8cc1-f0311422fbd3">

**Comment by LuluW8071** [comment, marked as answer]
It depends on hardware, data and model complexity u r training on. CPU generally takes longer time if there are lots of featured data and huge parameter in model. Try increasing batch size and decrease learning rate if it still persists OR Restart the kernel to refresh RAM and run again.

**Reply by SailSabnis** [reply]
Many thanks! will give it a try

**Reply by mrdbourke** [reply]
Hi @SailSabnis , 

If you are using a CPU for larger datasets, your model will likely take quite a bit longer to train.

GPU is best for larger datasets/models.

I'd only recommend CPU for smaller deep learning models/datasets.

**Reply by SailSabnis** [reply]
thinking of getting Colab pro (for a few months atleast) to avoid lot of time waste. Even 5 epochs take >25mins on CPU. It is frustrating.
